### DISCUSSION TITLE
something similar to fooocus v2's prompt expander?

### DISCUSSION THREAD

**Discussion by yesbroc** [Stability-AI/StableSwarmUI#107] [Ideas]
(i can't write prompts and fooocus v2 can apparently combine multiple expanders without quality loss)

**Comment by mcmonkey4eva** [comment]
per https://github.com/lllyasviel/Fooocus/discussions/117#raw they only have one expander, which is a small LLM to add words to the prompt.

The other styles are just prompt presets - in Swarm this is accessed via the "Presets" menu at the bottom, you can find a pack of good SDXL presets on <https://github.com/Stability-AI/StableSwarmUI/blob/master/docs/Presets.md> and you can of course import any other presets from elsewhere or make your own. You can probably dig into Fooocus's source to find the presets they use... the ones prefixed "SAI" would be StabilityAI's prompt presets which are in the pack I linked.

For LLM-based dynamic prompt improvements, please make a feature request on the [issues page](https://github.com/Stability-AI/StableSwarmUI/issues).

ps I recently made a Discord bot that combines a much more powerful LLM with SDXL, [source here](https://github.com/mcmonkey4eva/SimpleDiscordAIBot) or you can join [text gen webui discord](https://discord.gg/u27RhJECcW) or [r/StableDiffusion discord](https://discord.gg/4WbTj8YskM) to try the bot.
(Not the same as what fooocus does, but a good demo of what much more advanced LLM tooling can do)
